Source/WebKit/UIProcess/UserMediaPermissionRequestProxy.cpp

```diff
@@ -151,6 +151,14 @@ void UserMediaPermissionRequestProxy::promptForGetDisplayMedia(UserMediaDisplayC
 {
 #if ENABLE(MEDIA_STREAM) && PLATFORM(COCOA)
     ASSERT(m_manager && canPromptForGetDisplayMedia() && promptType != UserMediaDisplayCapturePromptType::UserChoose);
+
+    if (!isMainRunLoop()) {
+        callOnMainRunLoop([this, promptType]() {
```

Comment:
You'll need to protect |this| in the lambda, otherwise, I don't see what prevents |this| from getting destroyed by the time the main run loop processes the task.

Reply:
I personally would have used a ensureOnMainRunLoop() call at the call site in UIDelegate.mm. If the client is calling the completion handler off the main thread (which is a bug), then it is likely not just promptForGetDisplayMedia() that gets called off the main thread but also `allow()` and `deny()` which get called from the same completion handler.

Therefore, I suspect that this fix is only partial.